lab-said/model/toy.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+
+const debug = require('debug')('http:model-toy');
+const Child = require('../model/child');
+const mongoose = require('mongoose');
+
+const Toy = mongoose.Schema({
+  name: {type: String, required: true},
+  desc: {type: String, required: true},
+  child: {type: mongoose.Schema.Types.ObjectId, required: true, ref: 'child'},
+}, {timestamps: true});
+
+Toy.pre('save', function(next) {
+  debug('#pre-save Toy');
+
+  Child.findById(this.child)
+    .then(child => {
+      let toyIdSet = new Set(child.toys);
+      toyIdSet.add(this._id); // only going to to add if the _id isn't already present in the set
+      child.toys = Array.from(toyIdSet);
+      return child.save();
+    })
+    .then(next)
+    .catch(() => next(new Error('validation failed to create toy because child does not exist')));
+});
+
+Toy.post('remove', function(doc, next) {
+  debug('#post-remove Toy');
+
+  Child.findById(doc.child)
```

Comment:
This might need some work. When I deleted a toy it didnt remove the toy from the childs toy array when I checked the collection in my mongo shell.